src/util.ts

```diff
@@ -62,34 +51,66 @@ export function normalizeUri(uri: string): string {
 	return url.format(parts);
 }
 
-export function path2uri(root: string, file: string): string {
-	let ret = 'file://';
-	if (!strict && process.platform === 'win32') {
-		ret += '/';
-	}
+/**
+ * Produces a URI after resolving a relative path.
+ * @param root the path to resolve from: eg c:\Users\myself
+ * @param file the relative or absolute path to resolve
+ */
+export function resolvepath2uri(root: string, file: string): string {
+	const scheme = 'file://';
+	// if (!strict && process.platform === 'win32') {
+	// 	ret += '/';
+	// }
 	let p;
 	if (root) {
-		p = resolve(root, file);
+		p = platformSensitiveResolve(root, file);
 	} else {
 		p = file;
 	}
+	// add extra slash if drive letter is detected.
 	if (/^[a-z]:[\\\/]/i.test(p)) {
 		p = '/' + p;
 	}
 	p = p.split(/[\\\/]/g).map((val, i) => i <= 1 && /^[a-z]:$/i.test(val) ? val : encodeURIComponent(val)).join('/');
-	return normalizeUri(ret + p);
+	return normalizeUri(scheme + p);
+}
+
+/**
+ * From sindresorhus's file-url but with resolve and strict functionality removed
+ * @param path an absolute path
+ */
+export function path2uri(path: string): string {
+	if (typeof path !== 'string') {
+		throw new TypeError(`Expected a string, got ${typeof path}`);
+	}
+
+	// TODO: this should only accept absolute paths!
+
+	let pathName = path;
+	pathName = pathName.replace(/\\/g, '/');
+
+	// Windows drive letter must be prefixed with a slash
+	if (pathName[0] !== '/') {
+		pathName = `/${pathName}`;
+	}
+
+	// Escape required characters for path components
+	// See: https://tools.ietf.org/html/rfc3986#section-3.3
+	return encodeURI(`file://${pathName}`).replace(/[?#@]/g, encodeURIComponent);
```

Comment:
not sure what you mean by that, this is the path -> URI function, `uriToLocalPath` does URI ->path?